### PR TITLE
Dev conveniences

### PR DESCRIPTION
**Pull request by tonsky**
Adds:

- [x] `dev` to :extra-paths and `user` namespace that is loaded by default in dev
- [x] `clj-reload` integration to properly reload code and stateful systems without restarting
- [x] `eftest` to run tests and not be overwhelmed by output
- [x] `#p` macro to quickly track values (inspired by https://github.com/weavejester/hashp)
- [x] socket server at 6006 in case anyone needs it

### CODE REVIEW

**Comment by tonsky** [comment]
This is how eftest runner looks like

https://github.com/user-attachments/assets/73b8d04b-b888-471e-8d2a-88d595551ae6

File: server/src/tool.clj [hunk 1]
```diff
@@ -168,3 +170,29 @@
 (comment
   (start-portal!)
   (tap> {:hello [1 2 3]}))
+
+(def p-lock
```
Comment (stopachka) [inline server/src/tool.clj]: Nit: I would mark this private, and do `defn-` for p-impl, just to keep these all the under the `p` function.

File: server/src/tool.clj [hunk 2]
```diff
@@ -168,3 +170,29 @@
...
+                     (list? form)
+                     (= 'tool/p-impl (first form)))
+                  (TaggedLiteral/create 'p (nth form 3))
+                  form))
+              form)]
+    (locking p-lock
+      (println (str position " #p " form " => " (pr-str res))))
+    res))
+
+(defn p [form]
```
Comment (stopachka) [inline server/src/tool.clj]: Nit: would add a docstring about this, and a potential usage here.

**Review by stopachka — APPROVED** [review]
Heeck yeah!

**Pull request merged**